packages/perseus/src/util/snowman-utils.ts

```diff
@@ -0,0 +1,149 @@
+export enum WidgetType {
+    Categorizer = "categorizer",
+    CSProgram = "cs-program",
+    Definition = "definition",
+    Dropdown = "dropdown",
+    Explanation = "explanation",
+    Expression = "expression",
+    GradedGroup = "graded-group",
+    GradedGroupAnswerBar = "graded-group-answer-bar",
+    GradedGroupSet = "graded-group-set",
+    Grapher = "grapher",
+    Group = "group",
+    Iframe = "iframe",
+    Image = "image",
+    InputNumber = "input-number",
+    Interaction = "interaction",
+    InteractiveGraph = "interactive-graph",
+    InteractiveNumberLine = "interactive-number-line",
+    LabelImage = "label-image",
+    Matcher = "matcher",
+    Matrix = "matrix",
+    Measurer = "measurer",
+    Molecule = "molecule",
+    NumberLine = "number-line",
+    NumericInput = "numeric-input",
+    Orderer = "orderer",
+    Passage = "passage",
+    PassageRef = "passage-ref",
+    PassageRefTarget = "passage-ref-target",
+    Plotter = "plotter",
+    Protractor = "protractor",
+    PythonProgram = "python-program",
+    Radio = "radio",
+    Sorter = "sorter",
+    Table = "table",
+    Unit = "unit",
+    Video = "video",
+    VideoTranscriptLink = "video-transcript-link",
+}
+
+const QUESTION_WIDGETS = [
+    "numeric-input",
+    "input-number",
+    "radio",
+    "interactive-graph",
+    "interactive-number-line",
+    "categorizer",
+    "plotter",
+    "orderer",
+    "protractor",
+    "matcher",
+    "sorter",
+];
+
+// Regex for widget placeholders in a string.
```

Comment:
Thank you for these comments! They really help explain what we're trying to grep.